[Build Scripts] Enable OS X build when Xcode is located in a path con…

k8stone opened the pull request:
…taining spaces

Applied a combination of explicitly quoted paths and bash's string escaping facilities to allow a build to complete successfully.

jrose-apple commented on cmake/modules/AddSwift.cmake:
```diff
@@ -46,12 +46,12 @@ function(_add_variant_c_compile_link_flags
 
   list(APPEND result
-    "-isysroot" "${SWIFT_SDK_${sdk}_PATH}")
+    "-isysroot" "\"${SWIFT_SDK_${sdk}_PATH}\"")
```
The CMake changes here seem odd to me. This is a list and these paths are already quoted, which means at this point in time CMake knows what one element consists of. Can we accomplish this using the `VERBATIM` setting for all `add_custom_command_target` calls instead?

(I'd be fine with forcing this setting in `add_custom_command_target` itself, if it doesn't break anything.)

k8stone replied:
I'm not deeply attached to any particular strategy, just trying to unblock usage with one of my common configurations.  Alternatives that achieve the same end result are equally welcome.  There may be unnecessary changes here as I had to iterate a bit to get the build process to work at all when my Xcode path contains a space.  I can try to isolate which changes are strictly required and which are just aggressively paranoid.

gribozavr replied:
I have landed a patch that allowed me to bulid with an Xcode with spaces in the path.  The tests still had issues.  Could you pick up the work from there?  The next change should be in lit.cfg.

gribozavr commented on utils/build-script-impl:
```diff
@@ -241,11 +241,11 @@ function set_deployment_target_based_options() {
                         -DCMAKE_C_FLAGS="$(cmark_c_flags $deployment_target)"
                         -DCMAKE_CXX_FLAGS="$(cmark_c_flags $deployment_target)"
-                        -DCMAKE_OSX_SYSROOT:PATH="$(xcrun --sdk $xcrun_sdk_name --show-sdk-path)"
+                        -DCMAKE_OSX_SYSROOT:PATH=$(printf %q "$(xcrun --sdk $xcrun_sdk_name --show-sdk-path)")
```
These paths are also already quoted, I feel like the issue is in some other place.

k8stone replied:
While the paths are quoted here, they are subsequently used in other locations where they are inconsistently quoted.  Escaping whitespace in the path itself via printf %q resolved the related build issues I observed.

gribozavr commented:
This code is the issue, it should be fixed instead:

```
  # Convert variables to space-separated strings.
  string(REPLACE ";" " " c_compile_flags "${c_compile_flags}")
  string(REPLACE ";" " " link_flags "${link_flags}")
```

gribozavr commented:
https://github.com/apple/swift/commit/ba6643c7a4b851a131cd84616069f932747b5a7b is the first step.  The build passes, tests still have issues.

k8stone commented:
I'm content that Dmitri's approach above has addressed the issues this pull request was intended to solve.  Thanks!  Solving the remaining test issues would definitely be appreciated.
